src/streamlink/plugin/api/http_session.py

```diff
@@ -187,7 +187,7 @@ def request(self, method, url, *args, **kwargs):
                 if retries >= total_retries:
                     err = exception(f"Unable to open URL: {url} ({rerr})")
                     err.err = rerr
-                    raise err
+                    raise err from rerr
```

Comment:
Technically, this is the correct thing to do here, but the last time I had at look at this, I decided to ignore adding the reference of the original/parent exception, because this can change how the stack gets printed, and changing that behavior should ideally be done in a dedicated PR.

The line above was added during the early Livestreamer days when the `HTTPSession` subclass was added in 936e66dd90f67377451be27951a7d5553a09088c. I'm not sure, but I believe `raise from` was already a thing in Python 2.x, so this shouldn't have been added.

The whole implementation of the `HTTPSession.request()` method is old and pretty bad, so I really don't want to touch any of this.

Maybe I should `raise from None` here, to not introduce any changes.

Reply:
Related to #5047, btw

Reply:
Changed to `raise from None`. Don't want to introduce any changes.